### PR TITLE
Feature/tec 3465 event status

### PR DESCRIPTION
**Pull request by jesseeproductions**
Adds the backend coding to handle the event status includes:

- Service Provider
- Meta Support
- Classic Editor Support
- Migration of reason field to a single field
- Automated Tests

Artifact: https://cloudup.com/c5w99UdhkSX

_Ref:_ [TEC-3465](https://theeventscalendar.atlassian.net/browse/TEC-3465)
_Ref:_ [TEC-3466](https://theeventscalendar.atlassian.net/browse/TEC-3466)
_Ref:_ [TEC-3468](https://theeventscalendar.atlassian.net/browse/TEC-3468)

### CODE REVIEW

File: src/Tribe/Event_Status/Event_Status_Provider.php [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+
+/**
+ * Class Event_Status_Provider
+ *
+ * @since   TBD
+ *
+ * @package Tribe\Events\Event_Status
+ */
+class Event_Status_Provider extends \tad_DI52_ServiceProvider {
+	const DISABLED = 'TEC_EVENT_STATUS_DISABLED';
```
Comment (Camwyn) [inline src/Tribe/Event_Status/Event_Status_Provider.php:23]: Docblock, something like 
```suggestion
	/**
	 * The constant used to disable event status.
	 */
	const DISABLED = 'TEC_EVENT_STATUS_DISABLED';
```

File: src/Tribe/Event_Status/Models/Event.php [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+	}
+
+	/**
+	 * Retrieves an event's status reason.
+	 *
+	 * @since TBD
+	 *
+	 * @param WP_Post $event Event ID.
+	 *
+	 * @return string The event's status reason or empty string if none.
```
Comment (Camwyn) [inline src/Tribe/Event_Status/Models/Event.php:78]: Empty string also if default 'scheduled' status.

**Review by Camwyn — CHANGES_REQUESTED** [review]
Really just some docblock tweaks - looks great 😄

File: src/Tribe/Event_Status/Classic_Editor.php [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+
+		/**
+		 * Fires after the Metabox saved the data from the current request.
+		 *
+		 * @since TBD
+		 *
+		 * @param int $post_id The post ID of the event currently being saved.
+		 * @param array<string,mixed> The whole data received by the metabox.
+		 */
+		do_action( 'tribe_events_event_status_metabox_save', $post_id, $data );
```
[the remark this answers is not in the document]
Comment (jesseeproductions) [reply src/Tribe/Event_Status/Classic_Editor.php:206]: I added one before that seemed to be the easiest place.

**Review by Camwyn — APPROVED** [review]
🎸

**Pull request merged**